data.tf

```diff
@@ -67,12 +67,15 @@ locals {
   efs_file_system_id = lookup(var.efs_file_system, "file_system_id", "")
   efs_mount_point    = lookup(var.efs_file_system, "mount_point", "")
   efs_source_path    = lookup(var.efs_file_system, "source_path", "")
+  hostname           = length(var.alias) > 0 ? format("%s.%s", var.alias[0].hostname, var.alias[0].domain) : null
```

Comment:
Line 70 has a bug: hostname is optional, and should default to `var.name` per the docs, but here it is treated like a required variable, so it should be:

```terraform
length(var.alias) > 0 ? format("%s.%s", lookup(var.alias[0], "hostname", var.name), var.alias[0].domain) : null
```

Reply:
Code has been refactored per your suggestions. 

To be completely honest, a user omitting something as critical as a hostname or domain from the `alias` block is probably being foolish.